src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -100,5 +110,61 @@ public static class TagHelperOutputExtensions
                 tagHelperOutput.Attributes.Remove(attribute);
             }
         }
+
+        private static void CopyHtmlAttribute(
+            int allAttributeIndex,
+            TagHelperOutput tagHelperOutput,
+            TagHelperContext context)
+        {
+            var existingAttribute = context.AllAttributes[allAttributeIndex];
+            var copiedAttribute = new TagHelperAttribute
+            {
+                Name = existingAttribute.Name,
+                Value = existingAttribute.Value,
+                Minimized = existingAttribute.Minimized
+            };
+
+            // Move backwards through context.AllAttributes from the provided index until we find a familiar attribute
+            // in tagHelperOutput where we can insert the copied value in front of.
+            for (var i = allAttributeIndex - 1; i >= 0; i--)
+            {
+                var previouName = context.AllAttributes[i].Name;
+                var index = IndexOfFirstMatch(previouName, tagHelperOutput.Attributes);
```

Comment:
Don't we want the _last_ attribute with same `Name` here? For example,
``` html
  <element existing="value 1" existing="value 2" toBeCopied="value 3"/>
```
will currently be restored as
``` html
  <element existing="value 1" toBeCopied="value 3" existing="value 2"/>
```

Do tests cover this case i.e. `output.Attributes` contains duplicate attributes one of which should be placed just before the copied attribute?